### ISSUE TITLE
When the code was released?

### ISSUE THREAD

**Issue by QWTforGithub** [Say2L/FSHNet#1]
Hello, thank you for your work! When will the code be released?

**Comment by OrangeSodahub** [comment]
@Say2L Four months passed, where is the code?

**Comment by QWTforGithub** [comment]
> > [@Say2L](https://github.com/Say2L) Four months passed, where is the code?
> 
> The paper of arXiv version is available now. The code will be open sourced in the next few days.

Thank you. Looking forward to your code release.

**Comment by Say2L** [comment]
> > > [@Say2L](https://github.com/Say2L) Four months passed, where is the code?
> > 
> > 
> > The paper of arXiv version is available now. The code will be open sourced in the next few days.
> 
> Can you tell me which version of the waymo dataset you are using?

The version of Waymo dataset we used is v1.2

**Comment by QWTforGithub** [comment]
> > > > [@Say2L](https://github.com/Say2L) Four months passed, where is the code?
> > > 
> > > 
> > > The paper of arXiv version is available now. The code will be open sourced in the next few days.
> > 
> > 
> > Can you tell me which version of the waymo dataset you are using?
> 
> The version of Waymo dataset we used is v1.2



> > > > [@Say2L](https://github.com/Say2L) Four months passed, where is the code?
> > > 
> > > 
> > > The paper of arXiv version is available now. The code will be open sourced in the next few days.
> > 
> > 
> > Can you tell me which version of the waymo dataset you are using?
> 
> The version of Waymo dataset we used is v1.2

Thanks for your reply. However, when I tested it, it seemed to produce an infinite loop. Is this correct?

![Image](https://github.com/user-attachments/assets/2d6755a9-cfaf-4c8f-a190-ab91a8791ec4)

**Comment by Say2L** [comment]
> > > > > > [@Say2L](https://github.com/Say2L) Four months passed, where is the code?
> > > > > 
> > > > > 
> > > > > The paper of arXiv version is available now. The code will be open sourced in the next few days.
> > > > 
> > > > 
> > > > Can you tell me which version of the waymo dataset you are using?
> > > 
> > > 
> > > The version of Waymo dataset we used is v1.2
> > 
> > 
> > > > > > [@Say2L](https://github.com/Say2L) Four months passed, where is the code?
> > > > > 
> > > > > 
> > > > > The paper of arXiv version is available now. The code will be open sourced in the next few days.
> > > > 
> > > > 
> > > > Can you tell me which version of the waymo dataset you are using?
> > > 
> > > 
> > > The version of Waymo dataset we used is v1.2
> > 
> > 
> > Thanks for your reply. However, when I tested it, it seemed to produce an infinite loop. Is this correct?
> > ![Image](https://github.com/user-attachments/assets/2d6755a9-cfaf-4c8f-a190-ab91a8791ec4)
> 
> It seems that you encounter a NaN loss issue. We test the code under a two gpu environment and shutdown the amp. Other settings may result in the NaN loss.

You can also run the test.py by setting the 

> > > > > [@Say2L](https://github.com/Say2L) Four months passed, where is the code?
> > > > 
> > > > 
> > > > The paper of arXiv version is available now. The code will be open sourced in the next few days.
> > > 
> > > 
> > > Can you tell me which version of the waymo dataset you are using?
> > 
> > 
> > The version of Waymo dataset we used is v1.2
> 
> > > > > [@Say2L](https://github.com/Say2L) Four months passed, where is the code?
> > > > 
> > > > 
> > > > The paper of arXiv version is available now. The code will be open sourced in the next few days.
> > > 
> > > 
> > > Can you tell me which version of the waymo dataset you are using?
> > 
> > 
> > The version of Waymo dataset we used is v1.2
> 
> Thanks for your reply. However, when I tested it, it seemed to produce an infinite loop. Is this correct?
> 
> ![Image](https://github.com/user-attachments/assets/2d6755a9-cfaf-4c8f-a190-ab91a8791ec4)

By the way, the test process indeed takes a very long time.  If the losses are right, you can set the sampling interval of test to 5, which obtains similar results to the 1 interval setting, but runs much faster.

**Comment by Say2L** [comment]
By the way, the test process indeed takes a very long time. If the losses are right, you can set the sampling interval of test to 5, which obtains similar results to the 1 interval setting, but runs much faster.

**Comment by QWTforGithub** [comment]
> By the way, the test process indeed takes a very long time. If the losses are right, you can set the sampling interval of test to 5, which obtains similar results to the 1 interval setting, but runs much faster.

Thanks for your reply. I am currently attending the CVPR conference in the United States, and there is an 11-hour time difference between us. During training, I found that only 2 4090 GPUs can run the code due to shared memory. I checked the training loss and did not find NAN, which means the training was performed correctly. During the inference phase, I tried to use 1 GPU and batch size=2, and the above problem occurred. The final training loss is as follows:

![Image](https://github.com/user-attachments/assets/fe5da755-31d1-4d41-a092-9d34e1d879c1)

I'll try again according to your suggestion.

**Comment by QWTforGithub** [comment]
> > > Your final loss is a little higher than that in the log. What is your training batch size?
> > 
> > 
> > Thanks for your reply and I'm sorry for the late reply. Because I only have 4090 and no 80G GPU like H100, the Batch Size can only be set to 8 (2 4090 GPUs). I noticed that there would be problems with training with more than 3 GPUs. Can this be solved?
> 
> More GPUs with a total batch size of 16 should also be able to obtain similar results.

When more than 3 GPUs are used to execute the Waymo dataset, the dataset fails to load. Therefore, the code can only be executed on 2 GPUs. Can this problem be solved?

**Comment by Say2L** [comment]
> > > > Your final loss is a little higher than that in the log. What is your training batch size?
> > > 
> > > 
> > > Thanks for your reply and I'm sorry for the late reply. Because I only have 4090 and no 80G GPU like H100, the Batch Size can only be set to 8 (2 4090 GPUs). I noticed that there would be problems with training with more than 3 GPUs. Can this be solved?
> > 
> > 
> > More GPUs with a total batch size of 16 should also be able to obtain similar results.
> 
> When more than 3 GPUs are used to execute the Waymo dataset, the dataset fails to load. Therefore, the code can only be executed on 2 GPUs. Can this problem be solved?

I trained the model with both 4 GPUs and 2GPUs, but I didn't encounter that problem. Can you provide more detailed errors you encountered?
